lib/methods/veres-one/client.js

```diff
@@ -0,0 +1,318 @@
+/*!
+ * Copyright (c) 2018 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const url = require('url');
+const r2 = require('r2');
+// const https = require('https');
+
+const DEFAULT_LOCATION = 'ledger';
+const DEFAULT_MODE = 'test';
+
+class VeresOneClient {
+  constructor(options = {}) {
+    this.config = options.config || require('./veres-one.config');
+  }
+
+  /**
+   * Fetches a DID Document for a given DID.
+   *
+   * @param did {string}
+   * @param [mode] {string} One of 'dev'/'test'/'live'
+   * @param [hostname] {string} Optional hostname of ledger (falls back
+   *   to the default hostname for a particular mode)
+   *
+   * @returns {Promise<object>} Resolves to DID Document Fetch Result
+   */
+  async get({did, mode, hostname}) {
+    const ledgerAgent = await this.getAgent({mode, hostname});
+    const queryServiceUrl = url.parse(ledgerAgent.service.ledgerQueryService);
+    queryServiceUrl.search = `id=${did}`;
+    const didUrl = url.format(queryServiceUrl);
+
+    // const https = require('https');
+    // let agent;
+    // if(options.mode === 'dev') {
+    //   agent = new https.Agent({rejectUnauthorized: false});
+    // }
+    // _verbose(options, 'Retrieving remote DID Document...', {hostname});
+    // _debug(options, 'remote request', {url: didUrl});
+
+    const response = await r2({
+      url: didUrl,
+      method: 'POST',
+      // agent,
+      headers: {
+        'accept': 'application/ld+json, application/json'
+      }
+    }).response;
+
+    if(response.status !== 200) {
+      throw new Error('Failed to get DID Document; status=' + response.status);
+    }
+
+    const body = await response.json();
+
+    const result = {
+      found: true,
+      retry: false,
+      // timeMs,
+      // reqMs,
+      // start: reqOptions.start,
+      // retries: reqOptions.retries + 1,
+      type: 'LedgerDidDocument',
+      did,
+      // hostname,
+      doc: body.object,
+      meta: body.meta
+    };
+
+    return result;
+  }
+
+  /**
+   * Sends an operation to a Veres One ledger node.
+   *
+   * @returns {object} response
+   */
+  async send(options = {}) {
+    const ledgerAgent = await this.getAgent(options);
+    const {operation} = options;
+
+    const headers = options.headers ||
```

Comment:
is the intention that if one wants to change one of these headers, one must specify both?  Could be something like:
```
const headers = Object.assign({}, {...defaults}, options.headers);
```